hudi-common/src/main/java/org/apache/hudi/common/util/CompactionUtils.java

```diff
@@ -214,7 +216,7 @@ public static List<HoodieInstant> getPendingCompactionInstantTimes(HoodieTableMe
    */
   public static Option<Pair<HoodieTimeline, HoodieInstant>> getDeltaCommitsSinceLatestCompaction(
       HoodieActiveTimeline activeTimeline) {
-    Option<HoodieInstant> lastCompaction = activeTimeline.getCommitTimeline()
+    Option<HoodieInstant> lastCompaction = activeTimeline.getCommitTimeline().filter(s -> !s.getAction().equals(REPLACE_COMMIT_ACTION))
```

Comment:
replace_commit could also mean INSERT_OVERRIDE, INSERT_OVERRIDE_TABLE and DELETE_PARTITIONS. Can we consider those cases as well please

Reply:
here is the code snippet to detect if a replace commit instant refers to clustrering or not.
```
Option<Pair<HoodieInstant, HoodieClusteringPlan>> instantPlan = ClusteringUtils.getClusteringPlan(table.getMetaClient(), instant);
```
if instantPlan is non empty, then it refers to clustering.